Packs/GsuiteAuditor/Integrations/GsuiteAuditor/GsuiteAuditor.py

```diff
@@ -0,0 +1,344 @@
+
+
+@logger
+def test_module(client: GSuiteClient) -> str:
+    """
+    Performs test connectivity by valid http response
+
+    :param client: client object which is used to get response from api.
+
+    :return: raise ValueError if any error occurred during connection
+    :raises DemistoException: If there is any other issues while making the http call.
+    """
+    with GSuiteClient.http_exception_handler():
+        if ADMIN_EMAIL:
+            suffix = URL_SUFFIX.format('all', 'admin')
+            client.http_request(url_suffix=suffix, method='GET', params={'max_results': '1'})
+        else:
+            return_results("Please insert Admin Email parameter for the test to run")
+    return 'ok'
+
+
+''' COMMAND FUNCTIONS '''
+
+
+def activities_list_command(client: GSuiteClient, args: Dict[str, Any]) -> CommandResults:
+    """
+    Prints all activities in the G Suite instance.
+
+    :param client: client object which is used to get response from api
+    :param args: command arguments.
+
+    :return: CommandResults object with context and human-readable.
+    """
+    arguments = prepare_args_for_activities_list(args)
+    response = client.http_request(
+        url_suffix=URL_SUFFIX.format(args.get('user_key', 'all'), args.get('application_name')),
+        params=arguments)
+
+    # Readable Output
+    readable_items = prepare_readable_items(response.get('items', []))
+
+    readable_output = f'### Next Page Token: {response["nextPageToken"]}\n' if response.get("nextPageToken") else ''
+    readable_output += tableToMarkdown(
+        'Total Retrieved {}: {}'.format('Activities', len(response.get('items', []))),
+        readable_items,
+        ['time', 'applicationName', 'email', 'profileId', 'ipAddress', 'events']) if response.get(
+        'items') else 'No activities found for the given ' \
+                      'argument(s). '
+    # Context
+    context_outputs = prepare_output_for_activities_list(response)
+
+    return CommandResults(
+        outputs=context_outputs,
+        readable_output=readable_output,
+        raw_response=response
+    )
+
+
+@logger
+def fetch_incidents(client: GSuiteClient, first_fetch_time: str, fetch_limit: int, applications: Any) -> List[Dict]:
+    total_items = []
+    last_run_dict = demisto.getLastRun()
+    last_run = last_run_dict.get('last_run')
+    last_ids = last_run_dict.get('last_ids', [])
+    last_leftover_items = last_run_dict.get('last_leftover_items', [])
+    new_leftover_items = []
+    if not last_run:  # this is the first run
+        last_run = dateparser.parse(first_fetch_time).strftime(DATE_FORMAT)
+
+    end_time = create_end_time(last_run, "1 hours")
+
+    if len(last_leftover_items) != 0:
+        total_items = last_leftover_items[:fetch_limit]
+        new_leftover_items = last_leftover_items[fetch_limit:]
+
+    if len(total_items) < fetch_limit:
+        for app in applications:
+            total_items.extend(fetch_request(client, app, last_run, end_time))
```

Comment:
you are missing something here - consider the following:
for simplicity's sake, let's say we have one app fetching. If my max results in a 1000 but I get 2000 results in the last minute, I will miss the second 1000 since my response will only give me the first.
We should check that if the response holds a next page token and get all the items from it as well.

Only after you have all the activities from all the apps then you should sort.

Reply:
your'e right fixed it